File: abc326b/abc326b.ml
open Printf
open Scanf

let solve n =
  let l = [100;111;122;133;144;155;166;177;188;199;200;212;224;236;248;300;313;326;339;400;414;428;500;515;600;616;700;717;800;818;900;919] in
  List.find (fun x -> x >= n) l

let () =
  scanf "%d " solve |> printf "%d\n"

File: abc326b/abc326b.hs
import Control.Monad
import Data.List (find)

main :: IO ()
main = solve <$> readLn >>= print

solve :: Int -> Int
solve n = maybe 0 id $ find (>= n) l326
  where
    l326 = do
      c <- [1..9]
      x <- [0..9]
      guard $ c * x < 10
      return $ c * 100 + x * 10 + c * x
